src/EnergyPlus/HWBaseboardRadiator.hh

```diff
@@ -113,22 +113,42 @@ namespace HWBaseboardRadiator {
         Real64 AirOutletTempStd = 0.0;
         Real64 FracConvect = 0.0;
         Array1D<Real64> FracDistribToSurf;
-        Real64 TotPower = 0.0;
-        Real64 Power = 0.0;
-        Real64 ConvPower = 0.0;
-        Real64 RadPower = 0.0;
-        Real64 TotEnergy = 0.0;
-        Real64 Energy = 0.0;
-        Real64 ConvEnergy = 0.0;
-        Real64 RadEnergy = 0.0;
-        PlantLocation plantLoc{};
-        int BBLoadReSimIndex = 0;
-        int BBMassFlowReSimIndex = 0;
-        int BBInletTempFlowReSimIndex = 0;
-        int HeatingCapMethod = 0;           // - Method for heating capacity scaled sizing calculation (HeatingDesignCapacity, CapacityPerFloorArea,
-                                            // FracOfAutosizedHeatingCapacity)
-        Real64 ScaledHeatingCapacity = 0.0; // - scaled maximum heating capacity {W} or scalable variable of zone HVAC equipment, {-}, or {W/m2}
-        AccountingData Accounting;
+        Real64 TotPower;
+        Real64 Power;
+        Real64 ConvPower;
+        Real64 RadPower;
+        Real64 TotEnergy;
+        Real64 Energy;
+        Real64 ConvEnergy;
+        Real64 RadEnergy;
+        PlantLocation plantLoc;
+        int BBLoadReSimIndex;
+        int BBMassFlowReSimIndex;
+        int BBInletTempFlowReSimIndex;
+        int HeatingCapMethod;          // - Method for heating capacity scaled sizing calculation (HeatingDesignCapacity, CapacityPerFloorArea,
+                                       // FracOfAutosizedHeatingCapacity)
+        Real64 ScaledHeatingCapacity;  // - scaled maximum heating capacity {W} or scalable variable of zone HVAC equipment, {-}, or {W/m2}
+        Real64 ZeroBBSourceSumHATsurf; // used in baseboard energy balance
+        // Record keeping variables used to calculate QBBRadSrcAvg locally
+        Real64 QBBRadSource;       // Need to keep the last value in case we are still iterating
+        Real64 QBBRadSrcAvg;       // Need to keep the last value in case we are still iterating
+        Real64 LastSysTimeElapsed; // Need to keep the last value in case we are still iterating
+        Real64 LastTimeStepSys;    // Need to keep the last value in case we are still iterating
+        Real64 LastQBBRadSrc;      // Need to keep the last value in case we are still iterating
```

Comment:
There are several things here. 1 is that ZeroBBSourceSumHATsurf, QBBRad* variables were moved to the HWBB struct. 2 the Last* variables were also moved to the same struct without deleting the AccountingData struct. 3 this code used to initialize inline and here a constructor was added back in. We are moving away from constructors and initializing in line where the variable is decalred.

Reply:
Yes, I don't have a hard opinion on which method is better. I do like the grouping approach for like variables but this can be done either way.

Reply:
I went with the approach that was more like all the other models that do this--without the AccountingData struct.  So, I got rid of the redundant parameters and that struct and move all of the initializations to in line, eliminating the constructor completely.  Committing this momentarily...